sphinx/util/typing.py

```diff
@@ -28,3 +28,5 @@
 
 # title getter functions for enumerable nodes (see sphinx.domains.std)
 TitleGetter = Callable[[nodes.Node], text_type]
+
+Inventory = Dict[str, Dict[str, Tuple[str, str, str, str]]]
```

Comment:
Let me know if you'd like a comment above this line like the other types. I'm not sure what that text should be.

If the order of these definitions are important, let me know I can place this on whatever line you think is best.

Reply:
Please note `# inventory data on memory`.

Reply:
I'll do it after merge.